includes/Discovery.php

```diff
@@ -231,6 +248,57 @@ protected function get_schemaorg_metadata(): array {
 		return apply_filters( 'web_stories_story_schema_metadata', $metadata, $post );
 	}
 
+	/**
+	 * Get product schema data.
+	 *
+	 * @since 1.22.0
+	 *
+	 * @param array $products Array of products.
+	 * @return array
+	 */
+	protected function get_product_data( array $products ): array {
+		if ( ! $products ) {
+			return [];
+		}
+		$product_data = [];
+		foreach ( $products as $product ) {
+			$data = [
+				'@type'     => 'Product',
+				'brand'     => $product['productBrand'] ?? '',
+				'productID' => $product['productId'] ?? '',
+				'url'       => $product['productUrl'] ?? '',
+				'name'      => $product['productTitle'] ?? '',
+				'offers'    => [
+					[
+						'@type'         => 'Offer',
+						'price'         => $product['productPrice'] ?? 0,
+						'priceCurrency' => $product['productPriceCurrency'] ?? '',
+					],
+				],
```

Comment:
Looks like we can also add the `description` here with the `productDetails` field

```suggestion
				'@type'            => 'Product',
				'brand'            => $product['productBrand'] ?? '',
				'productID'        => $product['productId'] ?? '',
				'url'              => $product['productUrl'] ?? '',
				'name'             => $product['productTitle'] ?? '',
				'description'      => $product['productDetails'] ?? '',
				'offers'           => [
					[
						'@type'         => 'Offer',
						'price'         => $product['productPrice'] ?? 0,
						'priceCurrency' => $product['productPriceCurrency'] ?? '',
					],
				],
```

After that this should be good to go!

Reply:
Fixed.